redbot/core/config.py

```diff
@@ -321,11 +323,50 @@ def get_attr(self, item: str, default=None, resolve=True):
         return defaults
 
     async def set(self, value):
-        if not isinstance(value, dict):
-            raise ValueError(
-                "You may only set the value of a group to be a dict."
-            )
-        await super().set(value)
+        try:
+            self.validate_group_data(value)
+        except (TypeError, ValueError):
```

Comment:
This try/except is doing nothing, no matter what if `validate_group_data` throws an exception it will be raised.

Reply:
I tend to add these to make it clearer what exceptions a method might raise. However if you don't deem it necessary, I'm happy to remove it